Editor/Credentials/AwsCredentialsFactory.cs

```diff
@@ -11,5 +11,18 @@ public static AwsCredentials Create()
             return new AwsCredentials(textProvider, UnityLoggerFactory.Create(textProvider));
         }
     }
-}
 
+    public interface IAwsCredentialsFactory
+    {
+        internal AwsCredentials Create();
+        
+    }
+
+    public class AwsCredentialsFactoryWrapper : IAwsCredentialsFactory
```

Comment:
Is there a reason we're not using `AwsCredentialsFactory` to implement this interface? Usually adapters/wrappers are used when you're using an external library, but we're adding the wrapper to the same file as the original class